Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverridePublicInterface.h

```diff
@@ -26,6 +27,12 @@ namespace AzToolsFramework
             //! @return true if overrides are present on the given entity id.
             virtual bool AreOverridesPresent(AZ::EntityId entityId) = 0;
 
+            //! Gets the override type on the given entity id. The prefab that creates the overrides is identified
+            //! by the class implmenting this interface based on certain selections in the editor. eg: the prefab currently being edited.
+            //! @param entityId The id of the entity for which to get the override type.
+            //! @return an override type if an override exists on the given entity id.
+            virtual AZStd::optional<EntityOverrideType> GetOverrideType(AZ::EntityId entityId) = 0;
```

Comment:
I think this should just be called 'OverrideType' since it'll be possible to create a 'AddPrefab' override in the future too.

Also, 'RemovePrefab' should be one of the types supported as it is already a supported operation